lib/full-site-editing/templates.php

```diff
@@ -192,3 +192,88 @@ function set_unique_slug_on_create_template( $post_id ) {
 	}
 }
 add_action( 'save_post_wp_template', 'set_unique_slug_on_create_template' );
+
+/**
+ * Inject the skip-link.
+ */
+function gutenberg_inject_skip_link() {
+	global $template_html;
+
+	// Add the skip-link styles if needed.
+	if ( WP_Theme_JSON_Resolver::get_theme_data()->should_add_skip_link_styles() ) {
+		echo '<style id="skip-link-styles">';
+		echo '.skip-link.screen-reader-text{border:0;clip:rect(1px,1px,1px,1px);clip-path:inset(50%);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute !important;width:1px;word-wrap:normal !important;}';
+		echo '.skip-link.screen-reader-text:focus{background-color:#eee;clip:auto !important;clip-path:none;color:#444;display:block;font-size:1em;height:auto;left:5px;line-height:normal;padding:15px 23px 14px;text-decoration:none;top:5px;width:auto;z-index:100000;}';
+		echo '</style>';
+	}
+
+	// Get the selectors.
+	$selectors = WP_Theme_JSON_Resolver::get_theme_data()->get_skip_link_selectors();
+
+	// Sanity check.
+	if ( ! $selectors || ! $template_html ) {
+		return;
+	}
+
+	// Try to find the selector in $template_html and print the skip-link.
+	foreach ( $selectors as $selector ) {
+		if ( 0 === strpos( $selector, '#' ) ) {
+			$selector_no_hash = str_replace( '#', '', $selector );
+			if ( strpos( $template_html, 'id="' . $selector_no_hash . '"' ) || strpos( $template_html, "id='$selector_no_hash'" ) ) {
+				echo '<a class="skip-link screen-reader-text" href="' . esc_attr( $selector ) . '">' . esc_html__( 'Skip to content', 'gutenberg' ) . '</a>';
+				return;
+			}
+		}
+	}
+
+	// If we got this far, an element with the appropriate target ID was not located.
+	// Add a script to auto-generate the target and link based on the content we have.
+	add_action(
+		'wp_footer',
+		function() use ( $selectors ) {
```

Comment:
I'm not sure if we care about this yet, but AFAIK using an anonymous function as a hook callback would be a violation of Core coding standards.

Reply:
Yeah, I was thinking about that too but the coding standard exists because we want users to be able to remove the hook.
In this case they can do it using `remove_action( 'wp_body_open', 'gutenberg_inject_skip_link' );` so I was thinking it's OK... I can see though why someone would want to remove just that one and not its parent.

Moved to a separate function in https://github.com/WordPress/gutenberg/pull/28946/commits/540a5cd108ec0ee9c62afeb7aac8a288527f3b71 :+1: